File: removeDuplicatesFromSortedList.cpp
#include "leetcode.h"
class Solution {
public:
    ListNode *deleteDuplicates(ListNode *head) {
        if (head == NULL) {
            return head;
        }
        ListNode *current = head->next;
        ListNode *previous = head;
        while (current != NULL) {
            if (current->val == previous->val) {
                previous->next = current->next;
                ListNode *tmp = current;
                current = current->next;
                delete tmp;
            }
            else {
                previous = previous->next;
                current = current->next;
            }
        }
        return head;
    }
};

File: removeDuplicatesFromSortedList.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        // Start typing your Java solution below
        // DO NOT write main() function
        if (head == null) {
            return head;
        }
        ListNode p = head.next;
        ListNode previous = head;
        while (p != null) {
            if (p.val == previous.val) {
                previous.next = p.next;
            }
            else {
                previous = previous.next;
            }
            p = p.next;
        }
        return head;
    }
}
